crates/autopilot/src/fok_limit_orders/quoter.rs

```diff
@@ -171,12 +172,22 @@ impl LimitOrderQuoter {
     }
 }
 
-fn query_from(data: &OrderQuotingData) -> Query {
-    Query {
+fn balance_query(data: &OrderQuotingData) -> Option<Query> {
+    // Note that we skip balance checks for orders with pre-interactions. This
+    // notably includes EthFlow orders, as the WETH for the trade will get
```

Comment:
> Couldn't we still use the correct pre-interactions for all orders except ethflow?

This would require us to not split fee updates by buy/sell/amounts like we do right? This is a much bigger change :/

Reply:
I think it would only require us to make custom interactions part of the tuple we group open orders by.
I agree that this would be a way bigger change. Just wanted to mention it since I think it should be doable to do verified quotes and reuse them optimally across open orders.
Not a super high priority, though.

Reply:
I see... You are right, and this would be per user so in theory it should work... Let me see how complicated this to squeeze in here.

Reply:
I'm going to do this as a follow up PR, I can also clean up some other EthFlow balance check exceptions that we no longer needed.

Reply:
Tracked: https://github.com/cowprotocol/services/issues/1595